src/System.Management.Automation/engine/hostifaces/LocalPipeline.cs

```diff
@@ -1233,15 +1189,6 @@ internal class PipelineThread : IDisposable
         /// <summary>
         /// Creates the worker thread and waits for it to be ready
         /// </summary>
-#if CORECLR
-        internal PipelineThread()
-        {
-            _worker = new Thread(WorkerProc);
-            _workItem = null;
-            _workItemReady = new AutoResetEvent(false);
-            _closed = false;
-        }
-#else
         internal PipelineThread(ApartmentState apartmentState)
```

Comment:
Is it possible that this method can be called on Unix plats? If so, `_worker.SetApartmentState(apartmentState)` might throw.